WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java

```diff
@@ -404,169 +533,67 @@ private JSONObject serializeDotComParamsToJSONObject() throws JSONException {
-                    }
-                });
-    }
-
-    private void updateJetpackMonitorSettings() {
-        WordPress.getRestClientUtils().setJetpackMonitor(
-                mSite.getSiteId(), mJpSettings.monitorActive, new RestRequest.Listener() {
-                    @Override
-                    public void onResponse(JSONObject response) {
-                        mRemoteJpSettings.monitorActive = response.optBoolean("active");
-                        mJpSettings.monitorActive = mRemoteJpSettings.monitorActive;
-                        notifySavedOnUiThread(null);
-                    }
-                }, new RestRequest.ErrorListener() {
-                    @Override
-                    public void onErrorResponse(VolleyError error) {
-                        AppLog.w(AppLog.T.API, "Error updating Jetpack Monitor settings: " + error);
-                    }
-                });
-    }
-
-    private void saveJetpackSettings() {
-        updateJetpackMonitorSettings();
-        final Map<String, String> params = serializeJetpackParams();
-        if (params == null || params.isEmpty()) return;
-
-        WordPress.getRestClientUtils().setJetpackSettings(
-                mSite.getSiteId(), new RestRequest.Listener() {
-                    @Override
-                    public void onResponse(JSONObject response) {
-                        AppLog.d(AppLog.T.API, "Jetpack Settings saved remotely");
-                        mRemoteJpSettings.emailNotifications = mJpSettings.emailNotifications;
-                        mRemoteJpSettings.wpNotifications = mJpSettings.wpNotifications;
-                        notifySavedOnUiThread(null);
-                    }
-                }, new RestRequest.ErrorListener() {
-                    @Override
-                    public void onErrorResponse(VolleyError error) {
-                        AppLog.w(AppLog.T.API, "Error POSTing Jetpack site settings changes: " + error);
-                        notifySavedOnUiThread(error);
-                    }
-                }, params);
-    }
-
     private void deserializeJetpackRestResponse(SiteModel site, JSONObject response) {
         if (site == null || response == null) return;
         JSONObject settingsObject = response.optJSONObject("settings");
         mRemoteJpSettings.emailNotifications = settingsObject.optBoolean(JP_MONITOR_EMAIL_NOTES_KEY, false);
         mRemoteJpSettings.wpNotifications = settingsObject.optBoolean(JP_MONITOR_WP_NOTES_KEY, false);
     }
 
-    private Map<String, String> serializeJetpackParams() {
+    private @NonNull Map<String, String> serializeJetpackMonitorParams() {
         Map<String, String> params = new HashMap<>();
-        params.put(JP_MONITOR_EMAIL_NOTES_KEY, String.valueOf(mJpSettings.emailNotifications));
-        params.put(JP_MONITOR_WP_NOTES_KEY, String.valueOf(mJpSettings.wpNotifications));
+        if (mJpSettings.emailNotifications != mRemoteJpSettings.emailNotifications) {
```

Comment:
I personally would prefer not to make these checks because there is no guarantee that the remote settings is not changed since our last fetch/push and we don't really lose anything by updating these settings even if they are the same. That goes for everything in `serializeJetpackProtectAndSsoParams` too. I do see the logic behind it, so I am not saying we should change it, but just mentioning it as something to consider.